frontend/src/components/AddRetro.js

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { API_URL } from "../utils/constants";
+
+import retro from "../reducers/retro";
+
+import { WithContext as ReactTags } from "react-tag-input";
+import swal from "sweetalert";
+import styled from "styled-components";
+
+export const AddRetro = () => {
+  const [description, setDescription] = useState("");
+  const [participants, setParticipants] = useState([]);
+  const [userList, setUserList] = useState("");
+
+  const userId = useSelector((store) => store.user.userId);
+
+  const RefreshButton = () => {
+    window.location.reload("Refresh");
+  };
+
+  const dispatch = useDispatch();
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        description,
+        admin: userId,
+        participants,
+        active: true,
+      }),
+    };
+
+    fetch(API_URL("retros"), options)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.success) {
+          dispatch(retro.actions.setRetroId(data.response._id));
+        } else {
+          dispatch(retro.actions.setRetroId(null));
+        }
+      });
+    swal({
+      title: "Retro created successfully",
+      icon: "success",
+    }).then(RefreshButton);
+  };
+
+  useEffect(() => {
+    fetch(API_URL("users"))
+      .then((res) => res.json())
+      .then((data) => setUserList(data.response));
+  }, []);
+
+  // react tags functions
+  const KeyCodes = {
+    comma: 188,
+    enter: 13,
+  };
```

Comment:
This variable could be moved to `utils` directory or something like this - thanks to that, it can be reused in the future